core/src/main/java/io/grpc/internal/ManagedChannelImpl.java

```diff
@@ -589,7 +593,10 @@ public void updateRetainedTransports(Collection<EquivalentAddressGroup> addrs) {
     @Override
     public ClientTransport getTransport(final EquivalentAddressGroup addressGroup) {
       checkNotNull(addressGroup, "addressGroup");
-      TransportSet ts;
+      TransportSet ts = transports.get(addressGroup);
```

Comment:
Ack to your first answers, for the last part, I meant locking internal to  `TransportSet ts`  rather than `transports`.